src/couch/src/couch_compaction_daemon.erl

```diff
@@ -509,34 +509,47 @@ free_space(Path) ->
             length(filename:split(PathA)) > length(filename:split(PathB))
         end,
         disksup:get_disk_data()),
-    free_space_rec(abs_path(Path), DiskData).
+    {ok, Path0} = abs_path(Path),
```

Comment:
Maybe use `AbsPath` instead of `Path0`. Or at least `Path1` as `Path0` usually comes before Path.